### ISSUE TITLE
false positive using mapping unpack against class with no `__init__` method

### ISSUE THREAD

**Issue by BlueGlassBlock** [microsoft/pyright#4273]
**Describe the bug**
Passing an unpacked dict as kwargs to a class with no `__init__` constructor method is considered as an error

**To Reproduce**
```python
class A:
    def __init__(self) -> None:
        ...

class B:
    ...

A(**{}) # OK

B(**{}) # Expected no arguments to "B" constructor
```

**Expected behavior**
The code type checks.

```py
A(**{}) # OK

B(**{}) # OK
```

**Screenshots or Code**
See above

**VS Code extension or command-line**
Pylance v2022.12.11

**Additional context**
`mypy` checked the code above correctly.

**Comment by erictraut** [comment]
I agree this behavior is inconsistent. Pyright special-cases classes that provide no constructor and fall back on the `object` constructor.

You said in the bug report that this is a "false negative". Does that mean that you think both cases (`A` and `B`) should generate an error? Or did you mean that you think the error is a "false positive" and neither case should result in an error?

**Comment by BlueGlassBlock** [comment]
Oh, I'm thinking the latter one :P

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.283, which I just published. It will also be included in a future version of pylance.
